tests/test.py

```diff
@@ -201,6 +206,53 @@ def test_unique_pageview_fetching(self):
                          expected_pageview_counts)
 
 
+@contextmanager
+def TemporaryDirectory():
+    name = mkdtemp()
+    try:
+        yield name
+    finally:
+        shutil.rmtree(name)
+
+
+class TestCSVWriter(unittest.TestCase):
+    def test_default_filename(self):
+        start_date = datetime(2015, 2, 3, 9, 15, 30)
+        end_date = datetime(2015, 3, 6, 14, 27, 17)
+        writer = CSVWriter(start_date=start_date, end_date=end_date)
+
+        expected_filename = 'report_2015-02-03_2015-03-06.csv'
+        self.assertEqual(writer.output_filename, expected_filename)
+
+    def test_writing_csv(self):
+        datapoints = [
+            self.build_datapoint_with_counts('/path1'),
+            self.build_datapoint_with_counts('/path2'),
+        ]
+
+        expected_csv_lines = [
+            'uniquePageviews,problemReports,searchUniques,pagePath,_id,problemsPer100kViews,searchesPer100kViews',
+            '10,2,5,/path1,_path1,20000.0,50000.0',
+            '10,2,5,/path2,_path2,20000.0,50000.0',
+        ]
+
+        with TemporaryDirectory() as tempdir:
+            csv_file = NamedTemporaryFile(delete=False, dir=tempdir)
```

Comment:
Since you're guaranteeing to tear down the temporary directory after the test runs, there's no need to use a NamedTemporaryFile here, as well.  Instead, you can just do `os.path.join(tempdir, "file.csv")` or similar.